src/PBot/Issues/InvolvedIssueQuery.cs

```diff
@@ -0,0 +1,83 @@
+namespace PBot.Issues
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using Octokit;
+
+    internal class InvolvedIssueQuery
+    {
+        private static readonly Regex TaskForceRegex = new Regex(@"Task[\s-]?Force:\s*(.*)$", RegexOptions.Multiline | RegexOptions.IgnoreCase | RegexOptions.Compiled);
+        private static readonly Regex MentionRegex = new Regex(@"@[a-z0-9.-]+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        private readonly GitHubClient client;
+
+        public InvolvedIssueQuery(GitHubClient client)
+        {
+            this.client = client;
+        }
+
+        public async Task<IEnumerable<InvolvedIssue>> Perform(string username)
+        {
+            var repos = (await client.Repository.GetAllForOrg("Particular")).ToList();
+            var tasks = repos.Select(x => GetInvolvedIssuesForRepo(x, username));
+
+            return (await Task.WhenAll(tasks)).SelectMany(issue => issue);
+        }
+
+        private async Task<IEnumerable<InvolvedIssue>> GetInvolvedIssuesForRepo(Repository repo, string username)
+        {
+            var mentionedFilter = new RepositoryIssueRequest { State = ItemState.Open, Mentioned = username };
+            var assigneeFilter = new RepositoryIssueRequest { State = ItemState.Open, Assignee = username };
+            var creatorFilter = new RepositoryIssueRequest { State = ItemState.Open, Creator = username };
+
+            var tasks = new[] { mentionedFilter, assigneeFilter, creatorFilter }
+                .Select(async filter => await client.Issue.GetForRepository("Particular", repo.Name, filter));
+
+            var issues = (await Task.WhenAll(tasks))
+                .SelectMany(issue => issue)
+                .GroupBy(issue => issue.Url)
+                .Select(g => g.First());
```

Comment:
@wolfbyte I parallelised the three queries in the latest commit. How does this look?